app/views/views.py

```diff
@@ -0,0 +1,70 @@
+"""
+Author: Allan Katongole
+Date: 28th December 2018
+Views for the flask API
+of the iReporter challenge.
+---------------------------
+Most of the logic is run or
+implemented here.
+---------------------------
+Flask pluggable views will be
+used as opposed to procedural
+programming.
+"""
+
+from flask import jsonify, request, Response, json
+from flask.views import MethodView
+
+from app.models.models import Incident
+
+incidents=[]
+
+class IncidentAPI(MethodView):
+    
+    def get(self,id):
+        if id is None:
+            #return all redflags
+            redflags = []
+            for incident in incidents:
+                redflags.append(incident.convert_data())
+
+            if len(incidents) < 1:
+                return jsonify({"status":200, "message": "There are no incidents yet, Please create one to continue"})
+            return jsonify({"status":200, "data": redflags})
+        else:
+            #return specific red-flag if id is specified
+            unique_red_flag = [red_flag for red_flag in incidents if red_flag.__dict__['id'] == int(id)]
+            if not unique_red_flag:
+                return jsonify({"status":404, "error":"The redflag you requested doesnt exist"})
```

Comment:
If you do not explicitly define the trailing error, you will always receive a 200 for this request and yet the status in the body is 404

Reply:
Arthur, The 200 means the server correctly processed the request and in the body of the response there is a 404 explaining what happened to the request. There is a difference between server status codes and api status codes.